app/src/main/java/com/android/sample/model/recipe/CreateRecipeViewModel.kt

```diff
@@ -247,18 +278,34 @@ class CreateRecipeViewModel(private val repository: FirestoreRecipesRepository)
     val newUid = repository.getNewUid()
     require(newUid.isNotBlank()) { "Recipe ID must not be blank." }
     recipeBuilder.setId(newUid)
-
     try {
-      val recipe = recipeBuilder.build()
-      repository.addRecipe(
-          recipe,
-          onSuccess = {
-            _publishStatus.value = RECIPE_PUBLISHED_SUCCESS_MESSAGE
-            recipeBuilder.clear()
-          },
-          onFailure = { exception ->
-            _publishStatus.value = RECIPE_PUBLISH_ERROR_MESSAGE.format(exception.message)
-          })
+      if (_photo.value != null) {
+        repoImg.uploadImage(
+            newUid,
+            C.Tag.FIRESTORE_RECIPE_IMAGE_NAME,
+            ImageDirectoryType.TEST,
+            _photo.value!!.asImageBitmap(),
+            onSuccess = {
+              // Set the Image UID to the Builder
+              recipeBuilder.setPictureID(newUid)
+              val recipe = recipeBuilder.build()
+              repository.addRecipe(
+                  recipe,
+                  onSuccess = {
+                    _publishStatus.value = RECIPE_PUBLISHED_SUCCESS_MESSAGE
+                    recipeBuilder.clear()
+                  },
+                  onFailure = { exception ->
+                    _publishStatus.value = RECIPE_PUBLISH_ERROR_MESSAGE.format(exception.message)
+                  })
+            },
+            onFailure = {
+              // Throw an error if the image upload fails
+              throw IllegalArgumentException("Image upload failed.")
+            })
+      } else {
+        throw IllegalArgumentException("Image must not be blank.")
```

Comment:
Yes, I made two changes. First, if the photo is null, this will now throw a `NullPointerException` rather than the previous `IllegalArgumentException`. Second, when an `onFailure` callback is invoked, the error handling is now consistent across both callbacks by setting `_publishStatus.value = RECIPE_PUBLISH_ERROR_MESSAGE.format(exception.message)`. This approach ensures a uniform error-handling process.

Reply:
Thanks, @Ylanv! Moving to `NullPointerException` for a null photo makes sense, and I appreciate the uniform error-handling setup. Moving the error message to the constants file now, and will check the updated callback logic. Thanks for making those adjustments!